File: src/front/domains/compe/CompeService.ts
import axios from 'axios';
import Compe from './Compe';
import CompeEntry from './CompeEntry';
import CompeResponse from './CompeResponse';

export default class CompeService {

  async createCompe(compe: Compe) {
    return await axios.post(`/api/compe/new`, compe);
  }

  async getCompes(): Promise<CompeResponse[]> {
    return await axios.get(`/api/compe/my-compes`);
  }

  async getCompe(compeId: string): Promise<CompeResponse> {
    return await axios.get(`/api/compe/my-compe/${compeId}`);
  }

  async entryCompe(compe: CompeEntry) {
    return await axios.post(`/api/compe/entry/new`, compe);
  }
}

File: src/main/java/com/rakuriku/rakuriku/domain/service/compe/CompeService.java
package com.rakuriku.rakuriku.domain.service.compe;

import java.util.List;

import com.rakuriku.rakuriku.domain.model.compe.CompeEntity;
import com.rakuriku.rakuriku.domain.model.compe.CompeEntryManagementEntity;
import com.rakuriku.rakuriku.domain.model.compe.EntriesEntity;
import com.rakuriku.rakuriku.presentation.controller.compe.EntryEventDto;

public interface CompeService {
	CompeEntity createCompe(CompeEntity compe);

	List<CompeEntity> getCompes(String adminId);

	CompeEntity getCompe(String compeId);

	CompeEntryManagementEntity entryCompe(String compeId, EntriesEntity entriesEntity, List<EntryEventDto> list);

}
